addon/services/intl.js

```diff
@@ -17,117 +15,123 @@ import getDOM from '../-private/utils/get-dom';
 import { createIntl, createIntlCache, IntlErrorCode } from '@formatjs/intl';
 import flatten from '../-private/utils/flatten';
+import EventEmitter from 'eventemitter3';
+import { tracked } from '@glimmer/tracking';
 
-export default Service.extend(Evented, {
+export default class extends Service {
   /**
    * Returns an array of registered locale names
    *
    * @property locales
    * @public
    */
-  locales: computed('_intls', {
-    get() {
-      return Object.keys(this._intls);
-    },
-  }),
+  get locales() {
+    return Object.keys(this._intls);
+  }
 
   /** @public **/
-  locale: computed('_locale', {
-    set(_, localeName) {
-      const proposed = makeArray(localeName).map(normalizeLocale);
-
-      if (!isArrayEqual(proposed, this._locale)) {
-        set(this, '_locale', proposed);
-        cancel(this._timer);
-        this._timer = next(() => {
-          this.trigger('localeChanged');
-          this._updateDocumentLanguage(this._locale);
-        });
-      }
+  set locale(localeName) {
+    const proposed = makeArray(localeName).map(normalizeLocale);
+
+    if (!isArrayEqual(proposed, this._locale)) {
+      this._locale = proposed;
+
+      cancel(this._timer);
+      this._timer = next(() => {
+        this._ee.emit('localeChanged');
+        this._updateDocumentLanguage(this._locale);
+      });
+    }
+  }
 
-      return this._locale;
-    },
-    get() {
-      return this._locale;
-    },
-  }),
+  get locale() {
+    return this._locale;
+  }
 
   /**
    * Returns the first locale of the currently active locales
    *
    * @property primaryLocale
    * @public
    */
-  primaryLocale: computed.readOnly('locale.0'),
+  get primaryLocale() {
+    return this.locale[0];
+  }
 
   /** @public **/
-  formatRelative: createFormatterProxy('relative'),
+  formatRelative = createFormatterProxy('relative');
 
   /** @public **/
-  formatMessage: createFormatterProxy('message'),
+  formatMessage = createFormatterProxy('message');
 
   /** @public **/
-  formatNumber: createFormatterProxy('number'),
+  formatNumber = createFormatterProxy('number');
 
   /** @public **/
-  formatTime: createFormatterProxy('time'),
+  formatTime = createFormatterProxy('time');
 
   /** @public **/
-  formatDate: createFormatterProxy('date'),
+  formatDate = createFormatterProxy('date');
 
   /** @private **/
-  _locale: null,
+  @tracked _locale = null;
 
   /** @private **/
-  _timer: null,
+  _timer = null;
 
   /** @private **/
-  _formats: null,
+  _formats = null;
 
   /** @private **/
-  _formatters: null,
+  _formatters = null;
 
   /** @private */
-  _intls: null,
+  _intls = null;
+
+  /**
+   * @private
+   * @type {EventEmitter}
+   */
+  _ee = null;
 
-  _cache: createIntlCache(),
+  _cache = createIntlCache();
 
   /** @public **/
-  init() {
-    this._super(...arguments);
+  constructor() {
+    super(...arguments);
 
     const initialLocale = this.locale || ['en-us'];
-
+    this._ee = new EventEmitter();
```

Comment:
Looks like you can continue to use `Ember.Evented`, which is public api, by using `reopen` on the class.

Example: https://github.com/emberjs/ember.js/blob/ab785b81b9b3de771e05c4034e49d0e54795b189/packages/%40ember/-internals/routing/lib/services/router.ts#L514

Reply:
yeah but it seems like they're discouraging mixin as a whole. Besides exposing the service as a generic event emitter seems really broad for its API which is why I changed to `onLocaleChanged` instead.

Reply:
I'm fine with using a named fn like `onLocaleChanged` to register callbacks but think we could use `Ember.Evented` to achieve that so that we're not bundling an extra event emitter.

Reply:
eventemitter3 is tiny. Plus I'm going off https://ember-learn.github.io/ember-octane-vs-classic-cheat-sheet/ where they suggest not using the mixin pattern.